src/App.tsx

```diff
@@ -1,25 +1,39 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const startIndex = (currentPage - 1) * perPage;
+  const endIndex = Math.min(startIndex + perPage, items.length);
+  const itemsToShow = items.slice(startIndex, endIndex);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${startIndex + 1} - ${endIndex} of 42)`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={e => {
+              setPerPage(Number(e.target.value));
+              setCurrentPage(1);
+            }}
```

Comment:
it is bad practice set anonymous arrow function in jsx. better to make handler in component and pass reference on it to jsx.
```
const handleSelect = (e: ChangeEvent<HTMLSelectElement>) => {
  setPerPage(Number(e.target.value));
  setCurrentPage(1);
}
```
```suggestion
            onChange={handleSelect}
```